sdk/test-utils/recorder/README.md

```diff
@@ -1,9 +1,450 @@
+- `isRecordMode`, which is a shorthand for checking if the environment variable
+  `TEST_MODE` is set to `record`.
+- `isPlaybackMode`, which is a shorthand for checking if the environment
+  variable `TEST_MODE` is set to `playback`.
+- `setReplaceableVariables`, which will allow you to hide sensitive content
+  from the environment variables (more on that later).
+- `setReplacements`, which will allow you to hide sensitive content by doing
+  pattern matching in the output files.
+- `skipQueryParams`, since query parameters may contain sensitive information,
+  the array provided to method will signal what query parameters to remove from
+  the recordings.
+
+### Configuring your project
+
+Having the common recorder as a dependency means that you'll be able to start
+recording tests right away by using the exported method `record`. We'll get
+into the details further down this document. This function will do recordings,
+or will play back previous recordings, depending on an environment variable:
+`TEST_MODE`. If the environment variable `TEST_MODE` is empty, `record` (and most
+of the functions provided by test-utils-recorder) won't be doing anything. You'll need
+to set this environment variable to `record` to start recording, and then to
+`playback` to play the recordings back at your code.
+
+#### package.json scripts
+
+You can write scripts in your `package.json` to
+make it easier to switch from record mode to playback mode, on a meaningful context, as follows:
+
+    "integration-test:node": "mocha myNodeTests.js",
+    "unit-test:node": "TEST_MODE=playback npm run integration-test:node",
+    "test:node:record": "TEST_MODE=record npm run integration-test:node"
+
+
+Once your tests run, new files will be created in the `recordings/*` folder.
+These files will have names that are relative to the tests that you have.
+There might be cases in which the recordins get outdated with the test files, so you might also want to
+add a way to clear the recordings on your `package.json`, like the following one:
+
+    "clear-recordings": "rm -fr recordings",
+
+#### Karma configuration
+
+To make sure you're able to do record and playback on your browser tests, make
+sure to configure Karma properly.
+
+The recordings are separated between NodeJS recordings and browser recordings,
+so to use them on Karma, You'll at least need to add the recordings to your
+`files` array in your `karma.conf.js`, as follows:
+
+    files: [
+      // ... you might have other things here. Keep them.
+      "recordings/browsers/**/*.json"
+    ],
+
+Same goes on the `preprocessors` array:
+
+    preprocessors: {
+      // ... you might have other things here. Keep them.
+      "recordings/browsers/**/*.json": ["json"]
+    },
+
+You can also tell Karma to hide console.logs in during the playbacks in the browsers by adding
+the following configuration property:
+
+    browserConsoleLogOptions: {
+      terminal: process.env.TEST_MODE !== "record"
+    },
```

Comment:
You missed this **required** part
- Load `TEST_MODE` along with other variables

  ```javascript
  envPreprocessor: ["TEST_MODE"],
  ```